storage/reads/array_cursor.go

```diff
@@ -38,131 +51,16 @@ func newWindowAggregateArrayCursor(ctx context.Context, agg *datatypes.Aggregate
-		req: cursors.CursorRequest{
-			Ascending: asc,
-			StartTime: start,
-			EndTime:   end,
-		},
-	}
-
-	cc := cursorContext{
-		ctx: ctx,
-		req: &m.req,
-	}
-
-	m.cursors.i.cursorContext = cc
-	m.cursors.f.cursorContext = cc
-	m.cursors.u.cursorContext = cc
-	m.cursors.b.cursorContext = cc
-	m.cursors.s.cursorContext = cc
-
-	return m
-}
-
-func (m *arrayCursors) createCursor(seriesRow SeriesRow) cursors.Cursor {
-	m.req.Name = seriesRow.Name
-	m.req.Tags = seriesRow.SeriesTags
-	m.req.Field = seriesRow.Field
-
-	var cond expression
-	if seriesRow.ValueCond != nil {
-		cond = &astExpr{seriesRow.ValueCond}
-	}
-
-	if seriesRow.Query == nil {
-		return nil
-	}
-
-	var itr cursors.CursorIterator
-	var cur cursors.Cursor
-	for cur == nil && len(seriesRow.Query) > 0 {
-		itr, seriesRow.Query = seriesRow.Query[0], seriesRow.Query[1:]
-		cur, _ = itr.Next(m.ctx, &m.req)
-
-		switch c := cur.(type) {
-		case cursors.IntegerArrayCursor:
-			m.cursors.i.reset(c, seriesRow.Query, cond)
-			return &m.cursors.i
-		case cursors.FloatArrayCursor:
-			m.cursors.f.reset(c, seriesRow.Query, cond)
-			return &m.cursors.f
-		case cursors.UnsignedArrayCursor:
-			m.cursors.u.reset(c, seriesRow.Query, cond)
-			return &m.cursors.u
-		case cursors.StringArrayCursor:
-			m.cursors.s.reset(c, seriesRow.Query, cond)
-			return &m.cursors.s
-		case cursors.BooleanArrayCursor:
-			m.cursors.b.reset(c, seriesRow.Query, cond)
-			return &m.cursors.b
-		default:
-			panic(fmt.Sprintf("unreachable: %T", cur))
-		}
-	}
-	// if cursor is nil
-	return nil
-}
-
-func newSumArrayCursor(cur cursors.Cursor) cursors.Cursor {
-	switch cur := cur.(type) {
-	case cursors.FloatArrayCursor:
-		return newFloatArraySumCursor(cur)
-	case cursors.IntegerArrayCursor:
-		return newIntegerArraySumCursor(cur)
-	case cursors.UnsignedArrayCursor:
-		return newUnsignedArraySumCursor(cur)
-	default:
-		// TODO(sgc): propagate an error instead?
-		return nil
-	}
-}
-
-func newCountArrayCursor(cur cursors.Cursor) cursors.Cursor {
-	switch cur := cur.(type) {
-	case cursors.FloatArrayCursor:
-		return &FloatCountArrayCursor{FloatArrayCursor: cur}
-	case cursors.IntegerArrayCursor:
-		return &IntegerCountArrayCursor{IntegerArrayCursor: cur}
-	case cursors.UnsignedArrayCursor:
-		return &UnsignedCountArrayCursor{UnsignedArrayCursor: cur}
-	case cursors.StringArrayCursor:
-		return &StringCountArrayCursor{StringArrayCursor: cur}
-	case cursors.BooleanArrayCursor:
-		return &BooleanCountArrayCursor{BooleanArrayCursor: cur}
-	default:
-		panic(fmt.Sprintf("unreachable: %T", cur))
-	}
-}
-
 type cursorContext struct {
-	ctx   context.Context
-	req   *cursors.CursorRequest
-	itrs  cursors.CursorIterators
-	limit int64
-	count int64
-	err   error
+	ctx  context.Context
+	req  *cursors.CursorRequest
+	itrs cursors.CursorIterators
+	err  error
 }
 
 type multiShardArrayCursors struct {
-	ctx   context.Context
-	limit int64
-	req   cursors.CursorRequest
+	ctx context.Context
+	req cursors.CursorRequest
```

Comment:
Aggregation is now handled by WindowAggregateArrayCursor, so there's no need to calculate summary stats like count and limit here.